src/tools/kata-ctl/src/check.rs

```diff
@@ -222,6 +232,78 @@ pub fn check_official_releases() -> Result<()> {
     Ok(())
 }
 
+#[cfg(any(target_arch = "x86_64"))]
+pub fn check_kernel_module_loaded(module: &str, parameter: &str) -> Result<String, String> {
+    const MODPROBE_PARAMETERS_DRY_RUN: &str = "--dry-run";
+    const MODPROBE_PARAMETERS_FIRST_TIME: &str = "--first-time";
+    const MODULES_PATH: &str = "/sys/module";
+
+    // Partial check w/ modinfo
+    // verifies that the module exists
+    let output_modinfo = Command::new(MODINFO_PATH)
+        .arg(module)
+        .stdout(Stdio::piped())
+        .output()
+        .expect("failed to execute modinfo command");
```

Comment:
@dborquez - yes, as @amshinde says, we should be returning the error, not panicking:

```suggestion
        .output()?;
```

Reply:
What about adding a `.context("executing modinfo command")` ?